name: Yarn build and test
on: [push]

jobs:
  build:
    name: Test
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3

      - name: Set Node.js 22.x
        uses: actions/setup-node@v3
        with:
          node-version: 22.x

      - name: Run install
        uses: borales/actions-yarn@v4
        with:
          cmd: install --frozen-lockfile
      - name: Build production bundle
        uses: borales/actions-yarn@v4
        with:
          cmd: generate
      - name: Test the app
        uses: borales/actions-yarn@v4
        with:
          cmd: test

  prettier:
    name: Prettier Check
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3

      - name: Set Node.js 22.x
        uses: actions/setup-node@v3
        with:
          node-version: 22.x

      - name: Run install
        uses: borales/actions-yarn@v4
        with:
          cmd: install --frozen-lockfile

      - name: Run Prettier Check
        id: prettier-check
        uses: borales/actions-yarn@v4
        continue-on-error: true
        with:
          cmd: prettier:check
      
      - uses: mainmatter/continue-on-error-comment@v1 # Use this action to allow failure of prettier check since github does not have these feature
        with:
          repo-token: ${{ secrets.GITHUB_TOKEN }}
          outcome: ${{ steps.prettier-check.outcome }}
          test-id: Prettier Check